peps/pep-9999.rst

```diff
@@ -0,0 +1,465 @@
+                    a: int
+
+                type TDAsStr = <{K: str for K in KeyOf[TD]}>
+        * - Class equivalent
+          - .. code-block:: python
+
+                class TDAsStr(TypedDict):
+                    a: str
+
+* Changing the value type to ``str``, making the key not required:
+
+    .. list-table::
+        :widths: 25 75
+
+        * - Comprehension syntax
+          - .. code-block:: python
+
+                class TD(TypedDict):
+                    a: int
+
+                type TDAsNotRequiredStr = <{K: NotRequired[str] for K in KeyOf[TD]}>
+        * - Class equivalent
+          - .. code-block:: python
+
+                class TDAsNotRequiredStr(TypedDict):
+                    a: NotRequired[str]
+        * - Notes
+          - The :data:`~typing.NotRequired` type qualifier overrides the original ones
+            (in this case, we can assume :data:`~typing.Required` is an implicit qualifier on ``a``).
+
+* Making all keys not required, keeping the value type:
+
+    .. list-table::
+        :widths: 25 75
+
+        * - Comprehension syntax
+          - .. code-block:: python
+
+                class TD(TypedDict):
+                    a: int
+
+                type TDAsNotRequired = <{K: NotRequired[ValueOf[TD, K]] for K in KeyOf[TD]}>
+        * - Class equivalent
+          - .. code-block:: python
+
+                class TDAsNotRequired(TypedDict):
+                    a: NotRequired[int]
+        * - Notes
+          - The :data:`~typing.NotRequired` type qualifier overrides the original ones
+            (in this case, we can assume :data:`~typing.Required` is an implicit qualifier on ``a``).
+
+* Making all keys read only, wrapping the value type inside :class:`list`:
+
+    .. list-table::
+        :widths: 25 75
+
+        * - Comprehension syntax
+          - .. code-block:: python
+
+                class TD(TypedDict):
+                    a: NotRequired[int]
+
+                type TDAsReadOnlyList = <{K: ReadOnly[list[ValueOf[TD, K]]] for K in KeyOf[TD]}>
+        * - Class equivalent
+          - .. code-block:: python
+
+                class TDAsReadOnlyList(TypedDict):
+                    a: ReadOnly[NotRequired[list[int]]]
+        * - Notes
+          - Notice that :data:`~typing.NotRequired` is carried over, even if ``ValueOf[TD, K]`` is mapped in a
+            "nested" way.
+
+* Using a type alias to make every key not required:
+
+    .. list-table::
+        :widths: 25 75
+
+        * - Comprehension syntax
+          - .. code-block:: python
+
+                type Partial[T: TypedDict] = <{K: NotRequired[ValueOf[TD, K]] for K in KeyOf[T]}>
+        * - Class equivalent
+          - Not expressible
+
+* Using a type alias to select only certain keys:
+
+    .. list-table::
+        :widths: 25 75
+
+        * - Comprehension syntax
+          - .. code-block:: python
+
+                type Omit[T: TypedDict, K: KeyOf[T]] = <{P: ValueOf[T, P] for P in KeyOf[T] - K}>
```

Comment:
As mentioned in https://github.com/Viicos/peps/pull/1/files#r2312428163, this actually makes use of generic bounds.